docker.go

```diff
@@ -628,26 +628,30 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 			Since:      since,
 		}
 
-		ctx, cancel := context.WithTimeout(ctx, time.Second*5)
-		defer cancel()
-
 		r, err := c.provider.client.ContainerLogs(ctx, c.GetContainerID(), options)
 		if err != nil {
-			// if we can't get the logs, panic, we can't return an error to anything
-			// from within this goroutine
-			panic(err)
+			// if we can't get the logs, retry in one second.
+			if ctx.Err() != nil {
+				// context done.
+				return
+			}
+			c.logger.Printf("cannot get logs for container %q: %v", c.ID, err)
+			time.Sleep(1 * time.Second)
```

Comment:
Better use `time.After()` to allow cancellation when sleeping

```
			// if we can't get the logs, retry in one second.
			c.logger.Printf("cannot get logs for container %q: %v", c.ID, err)
			select {
				case <-ctx.Done():
					// context is done
					return
				case <-time.After(time.Second):
			}
```

If you really want to log when `err` is `nil` then you might want to add 1 more select case without the sleep above and put the  log in between those 1 selects.

Reply:
For the second log if `err == nil`, you mean simply this?
```golang
select {
default:
	c.logger.Printf("cannot get logs for container %q: %v", c.ID, err)
}
```